apidef/oas/middleware.go

```diff
@@ -475,3 +478,76 @@ type Header struct {
 	Name  string `bson:"name" json:"name"`
 	Value string `bson:"value" json:"value"`
 }
+
+func (ps Paths) fillCache(ep apidef.ExtendedPathsSet) {
+	advanceCacheConfig := &ep.AdvanceCacheConfig
+
+	for _, advCache := range *advanceCacheConfig {
+		if ps[advCache.Path] != nil {
+			plugins := ps[advCache.Path].getMethod(advCache.Method)
+			plugins.fillCache(advCache)
+
+		}
+	}
+}
+
+func (p *Plugins) fillCache(cacheMeta apidef.CacheMeta) {
+	if p.Cache == nil {
+		p.Cache = &CacheMiddleware{}
+	}
+	p.Cache.Fill(cacheMeta)
+}
+
+func (p *Plugins) extractCacheTo(ep *apidef.ExtendedPathsSet, path string, method string) {
+	advanceCacheConfig := ep.AdvanceCacheConfig
+	if advanceCacheConfig == nil {
+		advanceCacheConfig = []apidef.CacheMeta{}
+		newCache := apidef.CacheMeta{}
+		newCache.CacheOnlyResponseCodes = p.Cache.CacheResponseCodes
+		newCache.Disabled = !p.Cache.Enabled
+		newCache.CacheKeyRegex = p.Cache.CacheByRegex
+		newCache.Method = method
+		newCache.Path = path
+
+		advanceCacheConfig = append(advanceCacheConfig, newCache)
+		ep.AdvanceCacheConfig = advanceCacheConfig
+		return
+	}
+
+	founded := false
+	for _, advCache := range advanceCacheConfig {
+		if path == advCache.Path && method == advCache.Method && !founded {
+			founded = true
+			break
+		}
+	}
+	if !founded {
+		newCache := apidef.CacheMeta{}
+		newCache.CacheOnlyResponseCodes = p.Cache.CacheResponseCodes
+		newCache.Disabled = !p.Cache.Enabled
+		newCache.CacheKeyRegex = p.Cache.CacheByRegex
+		newCache.Method = method
+		newCache.Path = path
+
+		advanceCacheConfig = append(advanceCacheConfig, newCache)
+		ep.AdvanceCacheConfig = advanceCacheConfig
+	}
+}
+
+type CacheMiddleware struct {
```

Comment:
Let's rename it as `CachePlugin`, cache middleware is the one in the global level.